app/src/main/java/de/danoeh/antennapod/view/PagerIndicatorView.java

```diff
@@ -38,26 +40,29 @@ private void setup() {
         paint.setAntiAlias(true);
         paint.setStyle(Paint.Style.FILL);
 
-        int[] colorAttrs = new int[] { android.R.attr.textColorSecondary };
+        int[] colorAttrs = new int[]{android.R.attr.textColorSecondary};
         TypedArray a = getContext().obtainStyledAttributes(colorAttrs);
         circleColorHighlight = a.getColor(0, 0xffffffff);
         circleColor = (Integer) new ArgbEvaluator().evaluate(0.8f, 0x00ffffff, circleColorHighlight);
         a.recycle();
     }
 
-    public void setViewPager(ViewPager pager) {
-        numPages = pager.getAdapter().getCount();
-        pager.getAdapter().registerDataSetObserver(new DataSetObserver() {
+    public void setViewPager(ViewPager2 pager) {
+        numPages = pager.getAdapter().getItemCount();
+        pager.getAdapter().registerAdapterDataObserver(new RecyclerView.AdapterDataObserver() {
             @Override
             public void onChanged() {
-                numPages = pager.getAdapter().getCount();
+                numPages = pager.getAdapter().getItemCount();
                 invalidate();
             }
         });
-        pager.addOnPageChangeListener(new ViewPager.SimpleOnPageChangeListener() {
+        pager.registerOnPageChangeCallback(new ViewPager2.OnPageChangeCallback() {
             @Override
             public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
                 PagerIndicatorView.this.position = position + positionOffset;
+                if (ViewCompat.getLayoutDirection(pager) == ViewCompat.LAYOUT_DIRECTION_RTL) {
+                    PagerIndicatorView.this.position = numPages - 1 - PagerIndicatorView.this.position;
```

Comment:
Flips the pager dots for RTL languages, it can't be done once for the view as direction isn't specified at that time and should be done when the view is attached.